spec/support/shared_examples/services/chunkers/all_chunkers.rb

```diff
@@ -1,18 +1,16 @@
 RSpec.shared_examples "all chunkers" do
-  # Requires
-  # - `subject` to be instance of a chunker
-  # - the following `let` variables to be defined
-  #   - :text
-  #   - :text_type
+  let(:document) { create(:documentx) }
 
   describe "#chunk" do
-    let(:chunks) { subject.chunk(text) }
-
     it "returns Enumerable" do
+      parser = Parsers::CommonMark.new(document)
+      chunks = subject.chunk(parser)
```

Comment:
The convention in RSpec is not to define test objects right in examples, but to do them beforehand with `let`. This would remove the duplication between this example and the next.

Reply:
fixed